### PR TITLE
[4830][IMP] stock_picking_accounting_date: improve _compute_show_accounting…

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[4830](https://www.quartile.co/web#id=4830&cids=3&model=project.task&view_type=form)

### CODE REVIEW

File: stock_picking_accounting_date/models/stock_picking.py [hunk 1]
```diff
@@ -12,12 +12,23 @@ class StockPicking(models.Model):
...
+            move_ids = pick.move_ids.with_company(pick.company_id).filtered(
+                lambda move: move.product_id.detailed_type == "product"
+                and move.product_id.valuation == "real_time"
+            )
+            if move_ids:
+                move_source_locations = [move.location_id for move in move_ids]
+                move_dest_locations = [move.location_dest_id for move in move_ids]
+                locations = move_source_locations + move_dest_locations
+                if not self._check_internal_or_transit_location_type(locations):
+                    pick.show_accounting_date = True
```
Comment (yostashiro) [inline stock_picking_accounting_date/models/stock_picking.py:34]: We need to consider the combination of source and destination locations to judge whether there will be valuation records.

```suggestion
            valued_moves = pick.move_ids.with_company(pick.company_id).filtered(
                lambda move: move.product_id.detailed_type == "product"
                and move.product_id.valuation == "real_time"
                and not (move.location_id._should_be_valued() and move.location_dest_id._should_be_valued())
            )
            if valued_moves:
                pick.show_accounting_date = True
```

**Review by yostashiro — COMMENTED** [review]
Another approach we could consider is to make `accounting_date` always available.  There is certain need for users to be able to record the "actual" date of moves even when the move doesn't affect the valuation.  Let's talk about it.

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Can you please consolidate updates in https://github.com/qrtl/thc-oca/pull/44, closing this one.

**Comment by AungKoKoLin1997** [comment]
The changes was proposed in this commit https://github.com/qrtl/thc-oca/pull/44/commits/910bfd68e27ceb77b54b184aae2d231a3a4d596d and closing this PR.

**Pull request closed without merging**